include/lib/list.h

```diff
@@ -6,7 +6,8 @@
 
 /** Data structure for a generic list */
 struct list {
-    //TODO
+    /** The initial node of the list */
+    struct node* next;
```

Comment:
You should name this something like `head` instead of `next` to prevent confusion with `node.next`